### ISSUE TITLE
GSL integration

### ISSUE THREAD

**Issue by fredRos** [eos/eos#40]
- [x] Rename existing `integrate` to `integrate1D` to faciliate refactoring
- [x] Add interface for integrations with GSL for a given relative and absolute precision
```c++
template <typename Method_>
integrate(const std::function<double (const double&)>&, const double&a, const double& b, const Method_::Config& config);
integrate<GSL::QNG>(f, a, b, config);
```
where `Method_` is one of `GSL::QNG` (for Gauss-Kronrod) or `GSL::QAGS` (for adaptive
integration and unknown singular points). 

- [x] Convert all calls to our numerical integration to the new interface to GSL.
- [x] (Potentially) use Config class pattern
```c++
Config().relprecision(1e-3).absprecision(1e-5)....
```

**Comment by fredRos** [comment]
I saw instances of calls to `gsl_monte_miser` in three dimensions. Unfortunatly cubature-based multidimensional integration is not implemented in cuba but this might be much more efficient. Example libraries that I have used for this purpose are cuba and cubature. They typically require much smaller number of evaluations for a better precision in 3D.

**Comment by fredRos** [comment]
We still use the old integration routines for array integration...

**Comment by fredRos** [comment]
Cubature is coming in #60. This issue was solved with #44
